src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -414,8 +439,62 @@ class Config:
     ))
     async def do_update(self, data):
         """
-        Update LDAP Service Configuration.
-
+        `hostname` list of ip addresses or hostnames of LDAP servers with
+        which to communicate in order of preference. Failover only occurs
+        if the current LDAP server is unresponsive.
+
+        `basedn` specifies the default base DN to use when performing ldap
+        operations. The base must be specified as a Distinguished Name in LDAP
+        format.
+
+        `binddn` specifies the default bind DN to use when performing ldap
+        operations. The bind DN must be specified as a Distinguished Name in
+        LDAP format.
+
+        `anonbind` use anonymous authentication.
+
+        `ssl` establish SSL/TLS-protected connections to the LDAP server(s).
+        GSSAPI signing will be disabled on SSL/TLS-protected connections if
+        kerberos authentication is used.
+
+        `certificate` LDAPs client certificate to be used for certificate-
+        based authentication.
+
+        `validate_certificates` specifies whether to perform checks on server
+        certificates in a TLS session. If enabled, TLS_REQCERT demand is set.
+        The server certificate is requested. If no certificate is provided or
+        if a bad certificate is provided, the session is immediately terminated.
+        If disabled, TLS_REQCERT allow is set. The server certificate is
+        requested, but all errors are ignored.
+
+        `kerberos_realm` in which the server is located. This parameter is
+        only required for SASL GSSAPI authentication to the remote LDAP server.
+
+        `kerberos_principal` kerberos principal to use for SASL GSSAPI
+        authentication to the remote server. If `kerberos_realm` is specified
+        without a keytab, then the `binddn` and `bindpw` will be used to
+        perform to obtain the ticket necessary for GSSAPI authentication.
+
+        `timeout` specifies  a  timeout  (in  seconds) after which calls to
+        synchronous LDAP APIs will abort if no response is received.
+
+        `dns_timeout` specifies the timeout (in seconds) after which the
+        poll(2)/select(2) following a connect(2) returns in case of no activity.
```

Comment:
```suggestion
        poll(2)/select(2) following a connect(2) returns in case of no activity
```

I think there's an extra period there on the end?